resources/cmake/ConnextDdsAddSecurityArtifacts.cmake

```diff
@@ -0,0 +1,186 @@
+# (c) 2021 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+# No duplications, whole or partial, manual or electronic, may be made
+# without express written permission.  Any such copies, or revisions thereof,
+# must display this notice unaltered.
+# This code contains trade secrets of Real-Time Innovations, Inc.
+
+#[[.rst:
+.. _connextdds_add_security_artifacts:
+
+ConnextDdsAddSecurityArtifacts
+------------------------------
+
+Generate the artifacts needed for secure examples.
+
+``connextdds_add_security_artifacts()``
+
+It is based on:
+ - The OpenSSL configuration files in the resources/security/<ca> directory.
+
+ - The Governance and Permissions files in the resources/xml directory.
+
+So far this means:
+
+ - One ECDSA CA and two peers.
+
+ - RTPS and metadata protection kind set to ENCRYPT.
+
+ - Publishing and subscribing to any domain and topic is allowed, only for those
+   peers.
+
+#]]
+
+include(UseOpenSSL)
+
+
+function(connextdds_add_security_artifacts)
+    # Directory where the security artifacts are.
+    set(SECURITY_RESOURCES_PATH
+        ${SECURITY_RESOURCES_PATH}
+        "${CMAKE_MODULE_PATH}/../security"
```

Comment:
Since you want to make the `SECURITY_RESOURCES_PATH` related to this file, you should use `CMAKE_CURRENT_LIST_FILE` instead [documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_CURRENT_LIST_FILE.html#variable:CMAKE_CURRENT_LIST_FILE)

Reply:
`CMAKE_CURRENT_LIST_FILE` is the path to the file calling `connextdds_generate_security_artifacts`, in this case `<...>/network_capture/03_security/c/CMakeLists.txt`. That matches with what the documentation states in "Therefore the value of the variable inside a macro or function is the file invoking the bottom-most entry on the call stack, not the file containing the macro or function definition".

`SECURITY_RESOURCES_PATH` is where the source files for the security artifacts are (not where the artifacts will be generated). These are the cnf files that will be used to generate the certificates and the XML files that will be signed.
I will clarify this to make it less confusing.

Reply:
In any case, what I am currently doing is not alright, because `CMAKE_MODULE_PATH` is a list of paths.
I'll look into this.